material_components_ios.bzl

```diff
@@ -0,0 +1,49 @@
+"""Baze macros for building MDC component libraries."""
+
+load("@bazel_ios_warnings//:strict_warnings_objc_library.bzl", "strict_warnings_objc_library")
+
+def mdc_objc_library(
+    name,
+    copts = [],
+    **kwargs):
+  """Declare an Objective-C library with MDC's common warning flags."""
+  strict_warnings_objc_library(
+      name = name,
+      copts = copts,
+      **kwargs)
+
+def mdc_public_objc_library(
+    name,
+    deps = [],
+    extra_srcs = [],
+    sdk_frameworks = [],
+    visibility = ["//visibility:public"],
+    **kwargs):
+  """Declare a public MDC component as a Objective-C library according to MDC's conventions.
+
+  The conventions for an MDC component are:
+  - The public implementation lives inside of src.
+  - The private implementation lives inside of src/private.
+  - Headers are imported relative to src.
```

Comment:
What does this mean, "relative to src"? If I have a file at "src/private/foo.m" and it imports "src/private/foo.h", then it would contain `#import "private/foo.h"` ?

Reply:
Yep.

Reply:
Ack. That seems really awkward. Aren't includes usually either relative to the current file or just matched against files on the include path?

Reply:
Ah I misunderstood your original question - relative imports do work as expected and I agree that in that light this comment is somewhat misleading.

Reply:
The contents of these macros aren't Jeff's doing, I wrote them a while ago. Let's take that discussion offline so we keep the internal/external definitions in sync and Jeff doesn't have to guess at what I was trying to do. :)